app/src/main/java/com/android/halanvendordiscovery/domain/vendors/model/Page.kt

```diff
@@ -0,0 +1,9 @@
+package com.android.halanvendordiscovery.domain.vendors.model
+
+data class Page(val number: Int) {
+    fun increment(): Page = Page(number + 1)
+
+    companion object {
+        fun initial(): Page = Page(-1)
+    }
+}
```

Comment:
as we discussed before , this whole class could have been just a val

Reply:
In my opinion, it would be more beneficial to segregate all elements associated with the Page object within a dedicated class. This approach would facilitate future updates, particularly if there is a need to incorporate additional functions such as "previous" or a function that compares the count of page items based on future API updates.

Reply:
so the reason alone is correct and i see nothing wrong with the reason , yet my question is ,what does this class give me more that having a var named page of type Int as follows  var page = -1  and i'd be able to call page.inc(), i addition to it already having  page.dec() already implemented and accessible , while in the class i'ld have to re-implement such a "feature" , in other words , we're re-inventing the wheel here

Reply:
modifications have been committed